src/main/scala/viper/gobra/frontend/info/implementation/typing/ghost/GhostExprTyping.scala

```diff
@@ -335,6 +349,8 @@ trait GhostExprTyping extends BaseTyping { this: TypeInfoImpl =>
 
       // Others
       case PReceive(_) => false
+
+      case _: PPermission => true
```

Comment:
The children of a fractional permission could be impure (not now, but maybe in the future).